source/grid/grid_tools.cc

```diff
@@ -5998,18 +5998,6 @@ namespace GridTools
           if (perform_handshake)
             request_buffer = Utilities::pack(request_buffer_temp, false);
         },
-        [&](const unsigned int other_rank, std::vector<char> &recv_buffer) {
-          if (perform_handshake)
-            {
-              const auto other_rank_index = translate(other_rank);
-
-              recv_buffer =
-                Utilities::pack(std::vector<unsigned int>(
-                                  potential_owners_ptrs[other_rank_index + 1] -
-                                  potential_owners_ptrs[other_rank_index]),
-                                false);
```

Comment:
I meant to point this out before the review: This is the kind of thing one needed to do in these `prepare_for_answer()` functions. The *content* of `recv_buffer` didn't actually matter, the call here just makes sure that it has the right size. The call here is quite expensive, though: It creates a vector and serializes it. But the only thing that actually mattered was how large the resulting object is.